warehouse/source/source.go

```diff
@@ -143,6 +143,13 @@ func (m *Manager) process(ctx context.Context) error {
 	m.logger.Infow("starting source jobs processing")
 
 	for {
+		select {
```

Comment:
instead of moving the sleep above, we could have `processPendingJobs` return early(`nil`) if `len(pendingJobs) == 0`.
Just an alternative, either one's fine